driver-sync/src/test/functional/com/mongodb/client/unified/UnifiedTest.java

```diff
@@ -301,8 +305,22 @@ private void assertOutcome(final UnifiedTestContext context) {
         }
     }
 
+    private void assertOperationAndThrow(final UnifiedTestContext context, final BsonDocument operation, final int operationIndex) {
+        OperationResult result = executeOperation(context, operation, operationIndex);
+        assertOperationResult(context, operation, operationIndex, result);
+
+        if (result.getException() != null) {
+            throw (RuntimeException) result.getException();
+        }
+    }
+
     private void assertOperation(final UnifiedTestContext context, final BsonDocument operation, final int operationIndex) {
         OperationResult result = executeOperation(context, operation, operationIndex);
+        assertOperationResult(context, operation, operationIndex, result);
+    }
```

Comment:
I considered it but I'd prefer not to, for two reasons.

1. The return value is not needed in the other three places it's called, and it could create confusion that the return value is being ignored in those cases.
2. I don't like that the assertion becomes a side affect of an assignment statement i.e.

```
OperationResult result = assertOperation(context, operation, operationIndex);
```

it makes it easy to miss the intent.

So I don't think it's worth those costs to remove a line of code.